agent/containermetadata/write_metadata_unix_test.go

```diff
@@ -90,3 +90,35 @@ func TestWriteChmodFail(t *testing.T) {
 	assert.Error(t, err)
 	assert.Equal(t, expectErrorMessage, err.Error())
 }
+
+// TestSelinuxEnabledBindMode checks if data directory mounts in Z mode if Selinux enabled on docker server
+func TestSelinuxEnabledBindMode(t *testing.T) {
+	mockBinds := []string{}
+	mockEnv := []string{}
+	mockDataDirOnHost := ""
+	mockMetadataDirectoryPath := ""
+	mockSecurityOptions := []string{"selinux"}
+
+	binds, _ := createBindsEnv(mockBinds, mockEnv, mockDataDirOnHost, mockMetadataDirectoryPath, mockSecurityOptions)
+
+	actualBindMode := binds[0][len(binds[0])-2:]
+	expectedBindMode := ":" + bindMode
+
+	assert.Equal(t, actualBindMode, expectedBindMode)
+}
+
+// TestSelinuxDisabledBindMode checks if data directory is not mounted in Z mode if Selinux is not enabled on docker server
+func TestSelinuxDisabledBindMode(t *testing.T) {
+	mockBinds := []string{}
+	mockEnv := []string{}
+	mockDataDirOnHost := ""
+	mockMetadataDirectoryPath := ""
+	mockSecurityOptions := []string{""}
+
+	binds, _ := createBindsEnv(mockBinds, mockEnv, mockDataDirOnHost, mockMetadataDirectoryPath, mockSecurityOptions)
+
+	actualBindMode := binds[0][len(binds[0])-2:]
+	expectedBindMode := ":" + bindMode
+
+	assert.NotEqual(t, actualBindMode, expectedBindMode)
+}
```

Comment:
`TestSelinuxDisabledBindMode` and `TestSelinuxEnabledBindMode` can be made [table driven](https://github.com/golang/go/wiki/TableDrivenTests). 

An example in our codebase - https://github.com/aws/amazon-ecs-agent/blob/master/agent/engine/docker_task_engine_test.go#L171

Reply:
Created a table driven test for these two tests.